Plandomizer.py

```diff
@@ -962,6 +969,21 @@ def __init__(self, settings, src_dict=None):
 
         self.settings.__dict__.update(update_dict['_settings'])
         if 'settings' in self.src_dict:
+            for setting, choice in self.src_dict['settings'].items():
+                if not is_mapped(setting):
+                    raise TypeError('%s is not a valid setting. %s' % (repr(setting), build_close_match(setting, 'setting')))
+                info = get_setting_info(setting)
+                if type(choice) != info.type:
+                    raise TypeError('Supplied choice %s for setting %s is of type %s, expecting %s' % (repr(choice), repr(setting), repr(type(choice).__name__), repr(info.type.__name__)))
+                if not isinstance(choice, list) and choice not in info.choice_list:
+                    if setting == 'compress_rom' and choice == 'Temp':
+                        continue
+                    raise ValueError('%s is not a valid choice for setting %s. %s' % (repr(choice), repr(setting), build_close_match(choice, 'choice', info.choice_list)))
```

Comment:
Minor worry here: these will output the python representations rather than the json representations. For the most part that's totally fine, but True and False aren't capitalized in json :( Should we worry about that?

Reply:
I don't think that should be a worry, as the code should never reach here for a bool. Either the value for a boolean setting will be true/false, or it won't be and thus won't be a bool and will then be caught by the TypeError handler.

Reply:
Oh, right! Didn't think about the type checking.